src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/Assert.cs

```diff
@@ -344,6 +345,83 @@ public static void FileDoesNotExist(MSBuildResult result, params string[] paths)
             }
         }
 
+        public static Stream ContainsEmbeddedResource(string assemblyPath, string resourceName)
+        {
+            var stream = ExtractManifest(assemblyPath, resourceName);
+            Assert.NotNull(stream);
+
+            return stream;
+        }
+
+        public static void DoesNotContainEmbeddedResource(string assemblyPath, string resourceName)
+        {
+            var stream = ExtractManifest(assemblyPath, resourceName);
+            Assert.Null(stream);
+        }
+
+        private static unsafe Stream ExtractManifest(string path, string expectedResourceName)
+        {
+            using (var peStream = File.OpenRead(path))
+            {
+                using (var peReader = new PEReader(peStream))
+                {
+                    var mdReader = peReader.GetMetadataReader();
+
+                    foreach (var resourceHandle in mdReader.ManifestResources)
+                    {
+                        var resource = mdReader.GetManifestResource(resourceHandle);
+
+                        if (!resource.Implementation.IsNil)
+                        {
+                            continue; // resource is not embedded.
+                        }
+
+                        var resourceName = mdReader.GetString(resource.Name);
+                        if (!string.Equals(expectedResourceName, resourceName))
+                        {
+                            continue;
+                        }
+
+                        checked // arithmetic overflow here could cause AV
+                        {
+                            // Locate start and end of PE image in unmanaged memory.
+                            var block = peReader.GetEntireImage();
+                            Debug.Assert(block.Pointer != null && block.Length > 0);
+
+                            var peImageStart = block.Pointer;
+                            var peImageEnd = peImageStart + block.Length;
```

Comment:
@tmat - do we need to do this much math to read an embedded resource? It's for tests, and we don't want to load the assembly.